src/io.jl

```diff
@@ -17,6 +17,9 @@ Normally this involves defining:
 """
 abstract AbstractDataProvider
 
+type StubProvider <: AbstractDataProvider
```

Comment:
It is used in all functions, of the kind `function f(..., y::DataBatch, ...) = f(..., StubProvider(), y, )`. I thought, that construction `DataBatchProvider() = new(); DataBatchProvider(provider) = new(provider)` looks uglier, comparing to additional type. If better way to resolve this problem exists, or if `DataBatchProvider` syntax is ok, that's fine by me.